modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/DefaultExtensionsClient.java

```diff
@@ -0,0 +1,149 @@
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T, A extends Attributes> Publisher<Result<T, A>> executeAsync(String extension, String operation,
+                                                                        OperationParameters parameters)
+      throws MuleException {
+    OperationMessageProcessor processor = createProcessor(extension, operation, parameters);
+    Publisher<Event> publisher = processor.apply(just(getInitialiserEvent(muleContext)));
+    return from(publisher).map(e -> Result.<T, A>builder(e.getMessage()).build());
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T, A extends Attributes> Result<T, A> execute(String extension, String operation,
+                                                        OperationParameters parameters)
+      throws MuleException {
+    OperationMessageProcessor processor = createProcessor(extension, operation, parameters);
+    Event process = processor.process(getInitialiserEvent(muleContext));
+    return Result.<T, A>builder(process.getMessage()).build();
+  }
+
+  /**
+   * Creates a new {@link OperationMessageProcessor} for the required operation and parses all the parameters passed by
+   * the client user.
+   */
+  private OperationMessageProcessor createProcessor(String extensionName, String operationName, OperationParameters parameters) {
+    ExtensionModel extension = findExtension(extensionName);
+    OperationModel operation = findOperation(extension, operationName);
+    ConfigurationProvider config = parameters.getConfigName().map(this::findConfiguration).orElse(null);
+    OperationMessageProcessorFactory factory = new OperationMessageProcessorFactory(extension, operation, config, policyManager,
+                                                                                    toResolvers(parameters), muleContext, "");
+    try {
+      OperationMessageProcessor processor = factory.getOperationMessageProcessor();
+      processor.doStart();
+      processor.doInitialise();
+      return processor;
+    } catch (Exception e) {
+      throw new MuleRuntimeException(createStaticMessage("Could not create Operation Message Processor"), e);
+    }
+  }
+
+  private Map<String, Object> toResolvers(OperationParameters parameters) {
+    return parameters.getParameters().keySet().stream()
+        .collect(toMap(identity(), key -> new StaticValueResolver<>(parameters.getParameters().get(key))));
+  }
+
+  private ConfigurationProvider findConfiguration(String configName) {
+    return configurations.computeIfAbsent(configName, name -> extensionManager.getConfigurationProvider(configName)
+        .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("No configuration [" + configName + "] found"))));
+  }
+
+  private OperationModel findOperation(ExtensionModel extensionModel, String operationName) {
+    return operations.computeIfAbsent(operationName, operation -> ImmutableList.<OperationModel>builder()
```

Comment:
the key of the map cannot be the operationName since two different extensions can have operations with the same name. If you want to cache this, the key must be a Pair<ExtensionModel, OperationName>